test/es7/async-generator-apis.js

```diff
@@ -0,0 +1,216 @@
+            assert.areEqual(undefined, agfstrict.caller, "Get operation on 'caller' property returns undefined initially for a strict mode async function");
+            assert.areEqual(undefined, Object.getOwnPropertyDescriptor(agfstrict, "caller"), "No property descriptor for 'caller' initially for a strict mode async function");
+            assert.isTrue(delete agfstrict.caller, "Delete operation on 'caller' property returns true initially for a strict mode async function");
+
+            assert.areEqual(0, agfstrict.caller = 0, "Set operation on 'caller' creates new property with assigned value for a strict mode async function");
+            assert.isTrue("caller" in agfstrict, "Has operation on 'caller' property returns true now for a strict mode async function");
+            assert.areEqual(0, agfstrict.caller, "Get operation on 'caller' property returns property value now for a strict mode async function");
+            checkAttributes("agfstrict", agfstrict, "caller", { writable: true, enumerable: true, configurable: true });
+            assert.isTrue(delete agfstrict.caller, "Delete operation on 'caller' property still returns true for a strict mode async function");
+            assert.isFalse(agfstrict.hasOwnProperty("caller"), "'caller' property is gone for a strict mode async function");
+        }
+    },
+    {
+        name: "Async Generator function instances have %AsyncGeneratorFunctionPrototype% as their prototype and it has the correct properties and prototype",
+        body: function () {
+            async function* agf () {}
+            const asyncGeneratorFunctionPrototype = Object.getPrototypeOf(agf);
+
+            assert.areEqual(Function.prototype, Object.getPrototypeOf(asyncGeneratorFunctionPrototype), "%AsyncGeneratorFunctionPrototype%'s prototype is Function.prototype");
+
+            assert.isTrue(asyncGeneratorFunctionPrototype.hasOwnProperty("constructor"), "%AsyncGeneratorFunctionPrototype% has 'constructor' property");
+            assert.isTrue(asyncGeneratorFunctionPrototype.hasOwnProperty(Symbol.toStringTag), "%AsyncGeneratorFunctionPrototype% has [Symbol.toStringTag] property");
+
+            checkAttributes("%AsyncGeneratorFunctionPrototype%", asyncGeneratorFunctionPrototype, "constructor", { writable: false, enumerable: false, configurable: true });
+            checkAttributes("%AsyncGeneratorFunctionPrototype%", asyncGeneratorFunctionPrototype, Symbol.toStringTag, { writable: false, enumerable: false, configurable: true });
+
+            assert.areEqual("AsyncGeneratorFunction", asyncGeneratorFunctionPrototype[Symbol.toStringTag], "%AsyncGeneratorFunctionPrototype%'s [Symbol.toStringTag] property is 'asyncGeneratorFunction'");
+
+            assert.isTrue(asyncGeneratorFunctionPrototype.hasOwnProperty("prototype"), "%AsyncGeneratorFunctionPrototype% has a 'prototype' property");
+        }
+    },
+    {
+        name : "Async Generaotr function instances have the correct prototype object",
```

Comment:
```suggestion
        name : "Async Generator function instances have the correct prototype object",
```

Reply:
I've fixed this typo in the follow up PR #6172 thanks.